File: 22.06.29.java
import java.util.Arrays;
import java.util.Scanner;

public class math {
    static int cnt = 0;
    static int prime[] = new int[170000];
    static boolean IsPrime[] = new boolean[1000000];
    //Arrays.fill(IsPrime , true);
    public static void getPrime(){
        for(int i=0; i<1000000; i++){
            IsPrime[i] = true;
        }

        for(int i=2;i <=Math.sqrt(1000000); i++){
            if(IsPrime[i] == false)continue;
            for(int j=i*i; j<1000000; j+=i){
                IsPrime[j] = false;
            }
        }

        for(int i=2; i<1000000; i++){
            if(IsPrime[i]){
                prime[cnt] = i;
                cnt++;
            }
        }
    }

    public static void main(String args[]){
        getPrime();
        int N;
        int input[] = new int[100];
        int [][] input_prime= new int[100][170000];
        int sum_prime[] = new int[170000];
        for(int j=0; j<100; j++) {
            for (int i = 0; i < 170000; i++) {
                input_prime[j][i]=0;
            }
        }
        for(int i=0; i<100; i++){
            input[i] = 0;
        }
        //Arrays.fill(input_prime, 0);
        //Arrays.fill(sum_prime, 0);

        Scanner sc = new Scanner(System.in);
        N=sc.nextInt();
        for(int i =0; i<N; i++){
            input[i] = sc.nextInt();
        }

        for(int i =0; i<N; i++){
            for(int j=0; j<cnt; j++){
                while(input[i] % prime[j] == 0){
                    input[i]/=prime[j];
                    input_prime[i][j]++;
                    sum_prime[j]++;
                }
            }
        }
        int answer=1;
        int answer_cnt  =0;
        for(int i=0; i<cnt; i++){
            sum_prime[i] /=N;
        }
        for(int i=0; i<cnt; i++){
            if(sum_prime[i] >=1){
                answer *= Math.pow(prime[i], sum_prime[i]);
            }
        }

        for(int i=0; i<N; i++){
            for(int j=0; j<cnt; j++){
                if(sum_prime[j] - input_prime[i][j] > 0){
                    answer_cnt+=sum_prime[j] - input_prime[i][j];
                }
            }
        }
        System.out.print(answer+" "+ answer_cnt);
    }


}

File: 22.06.29.py
cnt = 0
real_prime = [0]*170000

prime = [True]*1000000

for i in range(2,int(len(prime)**(1/2))):
    if(prime[i] == False):continue
    for j in range(i*i, 1000000, i):
        prime[j]=False
for i in range(2,len(prime)):
    if(prime[i] == True):
        real_prime[cnt] = i
        cnt+=1
#N값 입력
N = int(input())
#N개의 값 입력
input2 = list(map(int, input().split()))
prime_input = [[0]*170000 for j in range(N)]
prime_count = [0]*170000
for i in range(N):
    for j in range(cnt):
        if real_prime[j]!=0:
            while input2[i] % real_prime[j] == 0:
                input2[i]/=real_prime[j]
                prime_input[i][j]+=1
                prime_count[j]+=1
answer=1
count = 0
for i in range(cnt):
    prime_count[i]/=N
for i in range(len(prime_count)):
    if prime_count[i] >= 1:
        answer *= real_prime[i] ** prime_count[i]
for i in range(N):
    for j in range(cnt):
        if int(prime_input[i][j] - prime_count[j]) < 0:
            count+=int(prime_count[j] - prime_input[i][j])
print(int(answer), int(count))

